lib/engine/step/g_1846/draft_distribution.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+require_relative '../base'
+
+module Engine
+  module Step
+    module G1846
+      class DraftDistribution < Base
+        attr_reader :companies
+
+        ACTIONS = %w[bid].freeze
+        ACTIONS_WITH_PASS = %w[bid pass].freeze
+
+        def setup
+          @companies = @game.companies.sort_by { @game.rand }
+          @choices = Hash.new { |h, k| h[k] = [] }
+          @draw_size = entities.size + 2
+        end
+
+        def pass_description
+          'Pass (Buy)'
+        end
+
+        def available
+          @companies.first(@draw_size)
+        end
+
+        def blank?(company)
+          company.name.include?('Pass')
+        end
+
+        def all_blank?
+          @companies.size < @draw_size && available.all? { |company| blank?(company) }
+        end
+
+        def only_one_company?
+          @companies.one? && !blank?(@companies[0])
+        end
+
+        def visible?
+          only_one_company?
+        end
+
+        def name
+          'Draft Round'
+        end
+
+        def description
+          'Draft Companies'
+        end
+
+        def finished?
+          all_blank? || @companies.empty?
+        end
+
+        def actions(_entity)
```

Comment:
you should only return actions if it's the user's turn, that way we don't have to check active entities anymore